core/trino-main/src/main/java/io/trino/server/protocol/spooling/SegmentResource.java

```diff
@@ -120,12 +123,12 @@ public HostAddress nextActiveNode()
     {
         List<InternalNode> internalNodes = ImmutableList.copyOf(nodeManager.getActiveNodesSnapshot().getAllNodes());
         verify(!internalNodes.isEmpty(), "No active nodes available");
-        return internalNodes.get(floorDiv(nextWorkerIndex.incrementAndGet(), internalNodes.size()))
+        return internalNodes.get(nextWorkerIndex.incrementAndGet() % internalNodes.size())
```

Comment:
this is not thread safe

Reply:
Why? The internalNodes is a copy of all active nodes

Reply:
Cool! I've made it use ThreadLocalRandom anyway